### ISSUE TITLE
[feature] Provide ability to suppress "skipped" dependencies messages when computing/displaying build requirements

### ISSUE THREAD

**Issue by marc-mw** [conan-io/conan#13518]
### What is your suggestion?

Currently, Conan 2.0 computes the transitive build-time dependencies of **_applications_** even if they are only being used as a build tool. Admittedly some of these dependencies might be needed if they are shared libraries, but for the ones that are statically linked into the application, it seems like the graph traversal could be terminated without needing to print out an informational message. Here's an example where zlib is being built with the Conan version of GCC, which itself has a lengthy list of dependencies. These are all noise from the perspective of simply building zlib:

> Graph root
>     cli
> Requirements
>     zlib/1.2.13#416618fa04d433c6bd94279ed2e93638 - Cache
> Build requirements
>     autoconf/2.71#c9c33c5eba84ace75c2609fc7182a686 - Cache
>     autoconf-archive/2021.02.19#26f25999387de5e6a9d7227ff3babe6e - Cache
>     automake/1.16.5#a30566899b45ec4cad5d500056e46c37 - Cache
>     binutils/2.38#04c643708b3bcfc7063eb005f5384e47 - Cache
>     cmake/3.25.1#3f033f447f7a59794066d3bd18056417 - Cache
>     flex/2.6.4#853349cfa9277b7708608a82a574d407 - Cache
>     gcc/10.3.0#c22960d72b0a408f97c8022cf6c78834 - Cache
>     gmp/6.2.1#8847660a551bf759454defa86eed09cf - Cache
>     gnu-config/cci.20210814#a59f719b7ed17c0ea1f991445f123ee4 - Cache
>     isl/0.24#a23ef985f95a54d1bec73aa0ab43f75a - Cache
>     libtool/2.4.7#9b3ddae55ab1f006744df695a3ce93dc - Cache
>     m4/1.4.19#c4b4789dd0785f632fc321cdd2662016 - Cache
>     mpc/1.2.0#7fe48d18befbd40b1702f6eb7920def4 - Cache
>     mpfr/4.1.0#57a05f57a42c58a7ad6990d708862081 - Cache
>     ninja/1.11.0#da75168088b60dc184c4bb6e66441012 - Cache
>     openssl/1.1.1s#6ac50ee9e58b08be28d7deb28dbd96b9 - Cache
>     zlib/1.2.13#416618fa04d433c6bd94279ed2e93638 - Cache
> 
> ======== Computing necessary packages ========
> Requirements
>     zlib/1.2.13#416618fa04d433c6bd94279ed2e93638:0b4324953d18bb5078b753b91f2488ae6cdfc53a#a9183dd207f32b930899602e62b9828b - Cache
> Build requirements
>     autoconf-archive/2021.02.19#26f25999387de5e6a9d7227ff3babe6e:da39a3ee5e6b4b0d3255bfef95601890afd80709#79dfc06fd0aba8e8a3c9191938537d7d - Skip
>     autoconf/2.71#c9c33c5eba84ace75c2609fc7182a686:da39a3ee5e6b4b0d3255bfef95601890afd80709#aad1124f372fa8de7669d24225cbc3a4 - Skip
>     automake/1.16.5#a30566899b45ec4cad5d500056e46c37:9a4eb3c8701508aa9458b1a73d0633783ecc2270#9719e51a6a62041af6a63e00eef35434 - Skip
>     binutils/2.38#04c643708b3bcfc7063eb005f5384e47:2b4f96dc00a236cef3f71a924654049326a61742#dc51eb78cc9ef7a56cce44fc90e3217e - Skip
>     cmake/3.25.1#3f033f447f7a59794066d3bd18056417:99ba5f8234b5341513c60c4f500b886d4a72f667#1737f30f71ed09ed42fbc2b920d6bc53 - Skip
>     flex/2.6.4#853349cfa9277b7708608a82a574d407:897e6fde718e72637d92b0b8ed929893a1976238#ccd518699a726fc8fe5a2602a75baf53 - Skip
>     gcc/10.3.0#c22960d72b0a408f97c8022cf6c78834:af41b17c3cdb0a6eefd3e0044e8a0de8ceb884cb#37211a10835cdfa01120ec2ff422d180 - Cache
>     gmp/6.2.1#8847660a551bf759454defa86eed09cf:e2042160f5f9e84f1f34511f201378289e063b83#905865120c15bb466d1eba2315377bf9 - Skip
>     gnu-config/cci.20210814#a59f719b7ed17c0ea1f991445f123ee4:da39a3ee5e6b4b0d3255bfef95601890afd80709#0d678e64f84cc730eaf93b400db0864f - Skip
>     isl/0.24#a23ef985f95a54d1bec73aa0ab43f75a:9ca3050f245a3bd66cd7a191c86d0d7ea547cebd#4dd63ed4b03335a34a16ff6767872f71 - Skip
>     libtool/2.4.7#9b3ddae55ab1f006744df695a3ce93dc:897e6fde718e72637d92b0b8ed929893a1976238#71b8943cdda9610b1e5e98dfea75245b - Skip
>     m4/1.4.19#c4b4789dd0785f632fc321cdd2662016:3593751651824fb813502c69c971267624ced41a#028e06b2ec8f151ba5ee47e640cc2c5c - Skip
>     mpc/1.2.0#7fe48d18befbd40b1702f6eb7920def4:95d790159428786d1c71779e5bee9788c659ea75#9df0f8a680ac8ce9363e5845c201c5d4 - Skip
>     mpfr/4.1.0#57a05f57a42c58a7ad6990d708862081:cc59a59152d2abf2d9dc21511523f4ee400e9670#707984b570266e24949080454582f397 - Skip
>     ninja/1.11.0#da75168088b60dc184c4bb6e66441012:3593751651824fb813502c69c971267624ced41a#0dc74834457ae37183757460a6996342 - Cache
>     openssl/1.1.1s#6ac50ee9e58b08be28d7deb28dbd96b9:896855b1e3b5961bfcc08e699116d9ed588cac00 - Skip
>     zlib/1.2.13#416618fa04d433c6bd94279ed2e93638:897e6fde718e72637d92b0b8ed929893a1976238#4b30748db76ec642513d35ebfecfc311 - Skip

A preferred display might look something like:

> Requirements
>     zlib/1.2.13#416618fa04d433c6bd94279ed2e93638 - Cache
> Build requirements
>     gcc/10.3.0#c22960d72b0a408f97c8022cf6c78834 - Cache
>     ninja/1.11.0#da75168088b60dc184c4bb6e66441012 - Cache
> ======== Computing necessary packages ========
> Requirements
>     zlib/1.2.13#416618fa04d433c6bd94279ed2e93638:0b4324953d18bb5078b753b91f2488ae6cdfc53a#a9183dd207f32b930899602e62b9828b - Cache
> Build requirements gcc/10.3.0#c22960d72b0a408f97c8022cf6c78834:af41b17c3cdb0a6eefd3e0044e8a0de8ceb884cb#37211a10835cdfa01120ec2ff422d180 - Cache
> ninja/1.11.0#da75168088b60dc184c4bb6e66441012:3593751651824fb813502c69c971267624ced41a#0dc74834457ae37183757460a6996342 - Cache

As this view shows what is "really" needed. Thanks,

P.S. As an aside, I am not sure why Conan realized it could skip CMake but still referenced Ninja from the cache, as both CMake and Ninja are called out in the [tool_requires] section of my host profile.

### Have you read the CONTRIBUTING guide?

- [X] I've read the CONTRIBUTING guide

**Comment by memsharded** [comment]
Hi @marc-mw 

Thanks for the suggestion.

We are not willing to introduce variability or configuration just to fine-tune the output. At most we would use the ``-vv``, ``-vvv`` to output different things. I'll check with the team, UX is very subjective, I prefer to explicitly see things, not sure what is the value of hiding that information.

**Comment by marc-mw** [comment]
Hi @memsharded,
My rationale here is that it's piercing an abstraction boundary. The notion that an executable was linked with a bunch of static libraries does not seem like something that is necessary or beneficial for downstream users of that tool or application to see. The other concern is that the volume of output as shown above dilutes the more important messages pertaining to the dependencies in effect at the time they are being displayed (that is the dependencies of the library or app being built) and the tools needed to build it (rather than, as in the example above, the tools needed to build the tools). Thanks,

**Comment by memsharded** [comment]
Understood, lets consider then the verbosity levels as a possible UI to control this output.

**Comment by franramirez688** [comment]
Closed by #13836 

If you want to see all the complete package references, you'll have to use the `-v` param explicitly in your command. Otherwise, you'll see something like this by default:

```
======== Computing necessary packages ========
app/1.0: Forced build from source
Requirements
    app/1.0#e622bb95f5703db8821d0d40434000d0:0d6ea6d022550d752f4feadc24de68a6116f4683 - Build
    pkg/1.0#85be0f54a32c5ee8b9c70c308f5c7084:0b2b3c6c1e4adad012025f459cdf5a0b7216babb#0be3f4aa351973fdcdc44dacc5868b9d - Cache
Test requirements
Skipped binaries
    ['mpfr/4.1.0', 'zlib/1.2.13`]
```
